src/CMakeLists.txt

```diff
@@ -77,6 +77,8 @@ if(WITH_SASL_SCRAM)
   list(APPEND sources rdkafka_sasl_scram.c)
 endif()
 
+list(APPEND sources rdkafka_sasl_oauthbearer.c)
```

Comment:
Since there is now a base64/openssl dependency, have this depend on WITH_SASL_OAUTHBEARER and add to the following files (search for and copy WITH_SASL_SCRAM, which also depends on base64):
```
$ git grep WITH_SASL_SCRAM
CMakeLists.txt:    set(WITH_SASL_SCRAM ON)
configure.librdkafka:        mkl_allvar_set WITH_SASL_SCRAM WITH_SASL_SCRAM y
packaging/cmake/config.h.in:#cmakedefine01 WITH_SASL_SCRAM
src/CMakeLists.txt:if(WITH_SASL_SCRAM)
src/CMakeLists.txt:  if(WITH_SASL_SCRAM)
src/CMakeLists.txt:    list(APPEND rdkafka_compile_definitions WITH_SASL_SCRAM)
src/CMakeLists.txt:  endif(WITH_SASL_SCRAM)
src/Makefile:SRCS_$(WITH_SASL_SCRAM) += rdkafka_sasl_scram.c
src/rdkafka_conf.c:#if WITH_SASL_SCRAM
src/rdkafka_conf.h:#if WITH_SASL_SCRAM
src/rdkafka_sasl.c:#if WITH_SASL_SCRAM
src/rdkafka_sasl.c:#if WITH_SASL_SCRAM
src/rdkafka_sasl_int.h:#if WITH_SASL_SCRAM
src/win32_config.h:#define WITH_SASL_SCRAM 1
```

Reply:
👍